### PR TITLE
Implement a patch for BSD service on OPNsense

### PR DESCRIPTION
**Pull request by johannrichard**
Although the current script will start the service on plain vanilla FreeBSD, that won't suffice for OPNsense which (also) requires that a `rc.d` script be enabled with a proper `<service>_enable="YES"` either in `/etc/rc.conf` or in a separate `/etc/rc.conf.d/<service>` file (https://docs.opnsense.org/development/backend/autorun.html):

*Part of the bootup process of OPNsense is probing the available `rc(8)` configuration files in `/etc/rc.conf.d/`, when a daemon is enabled, the system will call the regular `rc(8)` start command.*

With the implemented changes, `nextdns` will continue to work properly on plain vanilla FreeBSD but will also work on OPNSense. 

There is one caveat: If a user changes `nextdns_enable` to `NO` in the `rc(8)` configuration file, then `nextdns status` will return `running` because the `service nextdns status` command will return an exit code `0` even though it throws an error. Therefore, the service call should be `onestatus` instead of `status`, leading to the correct exit code.

[Edit]: I understood from [your comment](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=242752) that it's your intention to keep service configuration in the code, hence the extension for the `rc.conf.d` config in the code.

### CODE REVIEW

File: host/service/bsd/service.go [hunk 1]
```diff
@@ -17,7 +17,8 @@ import (
 type Service struct {
 	service.Config
 	service.ConfigFileStorer
-	Path string
+	RcDaemonPath string
+	RcConfigPath string
```
Comment (rs) [inline host/service/bsd/service.go:21]: Let's keep `Path` named that way so it stays homogeneous with other types of init.
RcConfigPath does not need to be in this struct as the template does not use it. Just use `rcConfPath(s.Name)` where needed.

File: host/service/bsd/service.go [hunk 2]
```diff
@@ -84,14 +102,20 @@ func (s Service) Restart() error {
...
 }
 
-var tmpl = `#!/bin/sh
+var rcConfTmpl = `# {{.Name}}
+# Set this to "NO" to disable 
+# the service but keep it installed
+{{.Name}}_enable="YES"
+`
+
+var rcDaemonTmpl = `#!/bin/sh
```
Comment (rs) [inline host/service/bsd/service.go:118]: Please don't rename this var (see above).

File: host/service/bsd/service.go (hunk 1 above)
Comment (johannrichard) [reply host/service/bsd/service.go]: Thanks, fixed.

**Comment by johannrichard** [comment]
> Looks good. Can you please also add support for OPNsense to the install.sh script?

Will do.

**Comment by rs** [comment]
Thx

**Pull request merged**